Add support for docker-archive: to skopeo list-tags (#1581)
containers/skopeo: docs/skopeo-list-tags.1.md

% skopeo-list-tags(1)

## NAME

> skopeo\-list\-tags - List tags in the transport-specific image repository or a local docker-archive file.

Reviewer: Non-blocking: How about something generic like “List image names in a transport-specific collection of images” ?

Author: this had done

## SYNOPSIS
**skopeo list-tags** [*options*] _source-image_

Return a list of tags from _source-image_ in a registry or a local docker-archive file.

>   _source-image_ name of repository to retrieve tag listing from or a local docker-archive file.

Reviewer 2: ```suggestion
  _source-image_ name of the repository to retrieve a tag listing from or a local docker-archive file.
```

Reviewer 2: I know it's old but would appreciate the touch-up if you make other changes.

>         "docker.io/myuser/myimage:v1.0"

Reviewer: Above, the text “Currently, only the Docker transport is supported.” needs to be updated (probably just deleted).

Author: this had done

        "docker.io/myuser/myimage@sha256:f48c4cc192f4c3c6a069cb5cca6d0a9e34d6076ba7c214fd0cc3ca60e0af76bb"

> ## Docker archive

Reviewer: This should follow format of the `docker://` paragraph above, an in particular precisely document the accepted syntax.

Author: this had done

>   **docker-archive:**_path_[**:**_docker-reference_]

Reviewer: Using the `[…docker-reference]` would AFAICS actually not work (it would not be a valid path the `Reader` could consume); so remove that part.

> An image is stored in the docker save formatted file.

Reviewer 3: ```suggestion
One or more images are stored in an `docker-archive` (see containers-transports(1)).
```
Plus add containers-transports(1) to the `# See also` section.

```

### Docker-archive Transport

```sh

> skopeo list-tags docker-archive:/tmp/busybox.tar.gz

Reviewer 4: Don't forget the prompt :)

> skopeo list-tags docker-archive:/root/busybox.tar.gz

Reviewer 4: Suggestion: include prompt on all sample commands. Since there's no reason to make this root-only, I would encourage you to use `$` and to change the path as well (to `/tmp/` instead of `/root`).

```

### Docker-archive Transport

To list the tags in a local docker-archive file:

```sh

> skopeo list-tags docker-archive:/tmp/busybox.tar.gz

Reviewer 2: Ed's suggestion I think...
```suggestion
$ skopeo list-tags docker-archive:/tmp/busybox.tar.gz
```

Author: done

{

>     "Repository": "",

Reviewer: Hum, should this field be filled in? Or hidden if empty?

I guess we can start with the latter, and fill the field in later if we find a need to do so.

Author: > Hum, should this field be filled in? Or hidden if empty?
> 
> I guess we can start with the latter, and fill the field in later if we find a need to do so.
may be there are more images in the file
```
$ ./bin/skopeo list-tags  docker-archive:docker-two-images.tar.xz
{
    "Repository": "",
    "Tags": [
        "example.com/empty:latest",
        "example.com/empty/but:different"
    ]
}
```

Reviewer: What I meant is that I think the output should be
```
$ ./bin/skopeo list-tags  docker-archive:docker-two-images.tar.xz
{
    "Tags": [
        "example.com/empty:latest",
        "example.com/empty/but:different"
    ]
}
```
without the pointless `Repository` line. That can be achieved using a `json:",omitempty"` annotation.
